### PR TITLE
O_DSYNC | O_DIRECT instead of O_SYNC + remove O_SYNC from MemMapPager

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Voron/Platform/Posix/AllowBuildOnWindows.cs [hunk 1]
```diff
@@ -508,7 +510,7 @@ public enum OpenFlags : int
 
         O_NOFOLLOW = 0x00020000,
         O_DIRECTORY = 0x00010000,
-        O_DIRECT = 0x00004000,
+        // O_DIRECT = 0x00004000,
```
Comment (ayende) [inline src/Voron/Platform/Posix/AllowBuildOnWindows.cs]: Why comment and not remove?

File: src/Voron/Platform/Posix/AllowBuildOnWindows.cs [hunk 2]
```diff
@@ -494,7 +494,9 @@ public enum OpenFlags : int
         O_TRUNC = 0x00000200,
         O_APPEND = 0x00000400,
         O_NONBLOCK = 0x00000800,
-        O_SYNC = 0x00001000,
+        O_SYNC = 1052672, // 0x00101000, // value directly from printf("%d", O_SYNC)
+        O_DSYNC = 4096, // 0x00001000, // value directly from printf("%d", O_DSYNC)
+        O_DIRECT = 16384, // 0x00004000, // value directly from printf("%d", O_DSYNC)
```
Comment (ayende) [inline src/Voron/Platform/Posix/AllowBuildOnWindows.cs:499]: You aren't actually using them, though

Comment (aviviadi) [inline src/Voron/Platform/Posix/AllowBuildOnWindows.cs:499]: I didn't commit the change by mistake. But I used it in the test.  Will be pushed in few minutes

File: src/Voron/Platform/Posix/PosixMemoryMapPager.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public PosixMemoryMapPager(int pageSize,string file, long? initialFileSize = nul
 
             PosixHelper.EnsurePathExists(_file);
 
-            _fd = Syscall.open(file, OpenFlags.O_RDWR | OpenFlags.O_CREAT | OpenFlags.O_SYNC,
+            _fd = Syscall.open(file, OpenFlags.O_RDWR | OpenFlags.O_CREAT,
```
Comment (ayende) [inline src/Voron/Platform/Posix/PosixMemoryMapPager.cs:23]: I'm feeling that this is bad. We have both PosixMemoryMapPager and PosixTempMemoryMapPager, why the difference?

It looks like we are using the temp just for pure memory stuff. But it looks like we can use it for things like scratch buffers in general.

Regardless, we shouldn't have O_SYNC here, btw

**Pull request merged**